brd-android/trade/src/main/java/com/fabriik/trade/ui/features/swap/SwapInputFragment.kt

```diff
@@ -0,0 +1,130 @@
+package com.fabriik.trade.ui.features.swap
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.activity.addCallback
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import com.breadwallet.tools.manager.BRSharedPrefs
+import com.fabriik.common.ui.base.FabriikView
+import com.fabriik.trade.R
+import com.fabriik.trade.databinding.FragmentSwapInputBinding
+import com.fabriik.trade.ui.customview.SwapCardView
+import kotlinx.coroutines.flow.collect
+import java.math.BigDecimal
+import androidx.core.view.isVisible
+
+class SwapInputFragment : Fragment(),
+    FabriikView<SwapInputContract.State, SwapInputContract.Effect> {
+
+    private lateinit var binding: FragmentSwapInputBinding
+    private val viewModel: SwapInputViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.fragment_swap_input, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = FragmentSwapInputBinding.bind(view)
+
+        with(binding) {
+            toolbar.setDismissButtonClickListener {
+                viewModel.setEvent(SwapInputContract.Event.DismissClicked)
+            }
+
+            cvSwap.setFiatCurrency(BRSharedPrefs.getPreferredFiatIso())
+            cvSwap.setCallback(object : SwapCardView.Callback {
```

Comment:
would it make sense to extract callback to a field/function to improve readability?